addons/sourcemod/scripting/include/nd_structures.inc

```diff
@@ -83,12 +101,49 @@ stock bool ND_IsStructFlameTurret(const char[] classname) {
 stock char[] GetRelayTowerName(int team)
 {
 	char name[32];
-	
+
 	switch (team)
 	{
 		case TEAM_CONSORT: Format(name, sizeof(name), STRUCT_REPEATER);
 		case TEAM_EMPIRE: Format(name, sizeof(name), STRUCT_RELAY);
 	}
-	
+
+	return name;
+}
+
+stock char[] GetStructureDisplayName(ND_Structures structure, bool uppercase=false)
+{
+	if (structure >= ND_StructCount || structure < ND_Command_Bunker)
+	{
+		ThrowError("Invalid structure specified to GetStructureDisplayName");
+	}
+
+	char name[MAX_STRUCT_DISPLAYNAME];
+	switch (structure)
+	{
+		case ND_Command_Bunker: Format(name, sizeof(name), DISPLAYNAME_BUNKER);
+		case ND_MG_Turret: Format(name, sizeof(name), DISPLAYNAME_MG_TURRET);
+		case ND_Transport_Gate: Format(name, sizeof(name), DISPLAYNAME_TRANSPORT);
+		case ND_Power_Plant: Format(name, sizeof(name), DISPLAYNAME_POWER_STATION);
+		case ND_Wireless_Repeater: Format(name, sizeof(name), DISPLAYNAME_REPEATER);
+		case ND_Relay_Tower: Format(name, sizeof(name), DISPLAYNAME_RELAY);
+		case ND_Supply_Station: Format(name, sizeof(name), DISPLAYNAME_SUPPLY);
+		case ND_Assembler: Format(name, sizeof(name), DISPLAYNAME_ASSEMBLER);
+		case ND_Armory: Format(name, sizeof(name), DISPLAYNAME_ARMORY);
+		case ND_Artillery: Format(name, sizeof(name), DISPLAYNAME_ARTILLERY);
+		case ND_Radar_Station: Format(name, sizeof(name), DISPLAYNAME_RADAR);
+		case ND_FT_Turret: Format(name, sizeof(name), DISPLAYNAME_FT_TURRET);
+		case ND_Sonic_Turret: Format(name, sizeof(name), DISPLAYNAME_SONIC_TURRET);
+		case ND_Rocket_Turret: Format(name, sizeof(name), DISPLAYNAME_ROCKET_TURRET);
+		case ND_Wall: Format(name, sizeof(name), DISPLAYNAME_WALL);
+		case ND_Barrier: Format(name, sizeof(name), DISPLAYNAME_BARRIER);
+	}
```

Comment:
This needs to go in an array and be indexed as I mentioned. We shouldn't be switching here.